mautic/mautic
# After remove contact next actions should not be executed for removed contact

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Suppose campaign have 2 events First for change campaign (Remove contact from this campaign) and second for another event now after execution of Change campaign action contact will be removed from campaign then second event should not be execute for contact because now that contact is not part of campaign but second event was also executing for removed contact. 


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a campaign segment 'Seg1' and add some contacts.
2. Create a Campaign 'Camp 2' having source as above created segment with any one simple action like 'adjust contact points'.
3. Create Campaign 'Camp 1' having source as above created segment with first Action 'Action 1' as Change Campaign.
In this change campaign action select remove contact from 'Camp 1/This Campaign' and add contact to 'Camp 2'.
4. Create Action 'Action 2' as Adjust contact points.
5. The Points from 'Action 2' got added to the contacts.

#### Steps to test this PR:
1. Create a campaign segment 'Seg1' and add some contacts.
2. Create a Campaign 'Camp 2' having source as above created segment with any one simple action like 'adjust contact points'.
3. Create Campaign 'Camp 1' having source as above created segment with first Action 'Action 1' as Change Campaign.
In this change campaign action select remove contact from 'Camp 1/This Campaign' and add contact to 'Camp 2'.
4. Create Action 'Action 2' as Adjust contact points.
5. Contact should be moved from 'Camp 1' to 'Camp 2' and as the contact is no longer in 'Camp 1' points should not get added of 'Action 2' to the contacts.

#### Other areas of Mautic that may be affected by the change:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2. 

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1.
